cxx-qt-build/src/lib.rs

```diff
@@ -143,139 +144,146 @@ fn extract_modules(file_content: &str, rs_path: &str) -> ExtractedModule {
     extracted
 }
 
-/// Write the generated cpp and h files for a qobject out to files
-fn write_qobject_cpp_files(obj: CppObject, snake_name: &str) -> Vec<String> {
-    let manifest_dir = manifest_dir();
-
-    let h_path = format!(
-        "{}/target/cxx-qt-gen/include/{}.h",
-        manifest_dir, snake_name
-    );
-    let cpp_path = format!("{}/target/cxx-qt-gen/src/{}.cpp", manifest_dir, snake_name);
-
-    let mut file = File::create(&h_path).expect("Could not create .h file");
-    write!(file, "{}", obj.header).expect("Failed to write .h file");
-
-    let mut file = File::create(&cpp_path).expect("Could not create .cpp file");
-    write!(file, "{}", obj.source).expect("Failed to write .cpp file");
-
-    vec![h_path, cpp_path]
+pub struct GeneratedCpp {
+    cxx_qt: Option<CppObject>,
+    cxx: cxx_gen::GeneratedCode,
+    module_ident: String,
```

Comment:
So the module ident is correct right now, however in the future we might support multiple objects being defined in one module and multiple modules in one file.

In this case, it's probably best just to match the input file name as the output filename and we'll use the namespace defined in the macro to avoid any collisions of similar names etc (i assume CXX does the same?).

So eg `lib.rs` -> `lib.h` or `my_object.rs` -> `my_object.h` etc.

Reply:
CXX appears to do them per file, so i think it's best to copy them. https://github.com/dtolnay/cxx/blob/9e0d3680be8cdc42a545467228fda096f34ce241/gen/src/mod.rs#L130  https://github.com/dtolnay/cxx/blob/9e0d3680be8cdc42a545467228fda096f34ce241/gen/src/write.rs#L17

Otherwise if we lean more on their generation later we could have mismatches of files (ours generating many files theirs generating one etc). But something to think about going forward as we continue evolving if this will cause issues and separate files are beneficial.

Reply:
You're right, cxx uses the Rust filename including `.rs` and adds `.h`/`.cpp`. So how about:

our generated files: filename.cxxqt.{h, cpp}
cxx generated files: filename.cxx.{h, cpp}

where `filename` includes the `.rs.`